presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1408,11 +1421,15 @@ VeloxQueryPlanConverterBase::toVeloxQueryPlan(
         if (!constantExpr->hasValueVector()) {
           setCellFromVariant(rowVector, row, column, constantExpr->value());
         } else {
-          auto columnVector = rowVector->childAt(column);
+          auto& columnVector = rowVector->childAt(column);
           columnVector->copy(constantExpr->valueVector().get(), row, 0, 1);
         }
       } else {
-        VELOX_FAIL("Expected constant expression");
+        // Evaluate the expression.
+        auto value = evaluateConstantExpression(expr);
```

Comment:
A noob question: given value nodes of sequence(1, 10000), the corresponding rows in ValuesNode will have size of 10000  with one RowExpression per each input value range from 1 to 10000? And then generate one input value at a time? Thanks!

Reply:
@xiaoxmeng ValueNode with sequence(1, 10000) expression represents a single row with a single column of type array. Presto planner evaluates sequence(1, 10000) and expands it to an array of 10K values. It then checks if serialized array size is too large and if so folds it back to sequence(1, 10000) expressions. Hence, sequence(1, 10) arrived to the worker as constant array, but sequence(1, 10000) arrives as an expression.